src/OpenSage.Game/Logic/Object/GameObject.cs

```diff
@@ -2031,7 +2031,7 @@ internal ObjectVeterancyHelper(GameObject gameObject, IGameObjectCollection game
 
     public bool GainExperience(int experience)
     {
-        if (_experienceSinkObjectId > 0)
+        if (_experienceSinkObjectId != ObjectId.Invalid)
```

Comment:
Can this just be IsValid? I think there might be other places where this could be changed as well.

Reply:
It definitely can! I think I added this check before I added `IsValid`. Find & replaced all of the simple cases I could find.